sheerwater_benchmarking/forecasts/gefs.py

```diff
@@ -0,0 +1,505 @@
+    dask_remote,
+    lon_base_change,
+    regrid,
+    shift_forecast_date_to_target_date,
+    target_date_to_forecast_date,
+)
+
+
+@dask_remote
+@cacheable(data_type='array',
+           cache_args=['variable', 'forecast_type',
+                       'run_type', 'time_group', 'grid'],
+           cache=False,
+           timeseries=['start_date', 'model_issuance_date'],
+           chunking={"lat": 300, "lon": 300, "lead_time": 46,
+                     "start_date": 29, "start_year": 29,
+                     "model_issuance_date": 1})
+def gefs_raw(start_time, end_time, variable, forecast_type,
+            run_type='average', time_group='weekly', grid="global0_25"):
+    """Raw GEFS forecast data.
+
+    Args:
+        start_time (str): The start date to fetch data for.
+        end_time (str): The end date to fetch.
+        variable (str): The weather variable to fetch.
+        forecast_type (str): The type of forecast to fetch. One of "forecast" or "reforecast".
+        run_type (str): The type of run to fetch. One of:
+            - average: to download the averaged of the perturbed runs
+            - perturbed: to download all perturbed runs
+            - [int 0-50]: to download a specific  perturbed run
+        time_group (str): The time grouping to use. One of: "daily", "weekly", "biweekly"
+        grid (str): The grid resolution to fetch the data at. One of:
+            - global0_25: 0.25 degree global grid
+    """
+    if grid != 'global0_25':
+        raise NotImplementedError(
+            "Only global 0.25 degree grid is implemented.")
+
+    # Open the datastore from dynamical.org
+    store = 's3://us-west-2.opendata.source.coop/dynamical/noaa-gefs-forecast/v0.1.0.zarr'
+
+    ds = xr.open_dataset(
+        store,
+        engine='zarr',
+        backend_kwargs={
+            'storage_options': {'anon': True},
+            'consolidated': True
+        }
+    )
+
+    # Select the right time period
+    ds = ds.sel(init_time=slice(start_time, end_time))
+
+    # Rename to match interface conventions
+    ds = ds.rename(
+        {
+            'init_time': 'start_date',
+            'latitude': 'lat',
+            'longitude': 'lon',
+            'ensemble_member': 'member'
+        }
+    )
+
+    # Handle different run types
+    if run_type == 'average':
+        # Take ensemble mean
+        ds = ds.mean(dim='member')
+    elif isinstance(run_type, int):
+        # Take specific ensemble member
+        ds = ds.isel(member=run_type)
+    elif run_type != 'perturbed':
+        raise ValueError(f"Invalid run_type: {run_type}")
+
+    # TODO: Select the right variable?
+    # Should that be done here caching each variable separately
+    # or in the gefs function caching all variables together?
+
+    # TODO: is forecast_type a relevant argument for this dataset?
+
+    # TODO: is time_group a relevant argument for this dataset?
+
+    # TODO: figure out what field conversions/aggregations need to be done
```

Comment:
I think we should roll up into daily data here? averaging the temperature and summing the precip, making each forecast a daily time series